tensorflow/compiler/mlir/python/mlir.cc

```diff
@@ -133,6 +148,87 @@ std::string ImportFunction(const std::string &functiondef_proto,
                                  status);
 }
 
+std::string ImportGraphDef(const std::string& proto,
+                           const std::string& pass_pipeline,
+                           bool show_debug_info, TF_Status* status) {
+  GraphDebugInfo debug_info;
+  GraphImportConfig specs;
+  return ImportGraphDefImpl(proto, pass_pipeline, show_debug_info, debug_info,
+                            specs, status);
+}
+
+std::string ImportGraphDef(const std::string& proto,
+                           const std::string& pass_pipeline,
+                           bool show_debug_info, absl::string_view input_names,
+                           absl::string_view input_data_types,
+                           absl::string_view input_data_shapes,
+                           absl::string_view output_names, TF_Status* status) {
+  std::vector<string> node_names = absl::StrSplit(input_names, ',');
+  std::vector<string> node_dtypes = absl::StrSplit(input_data_types, ',');
+  std::vector<string> node_shapes_str = absl::StrSplit(input_data_shapes, ':');
+  std::vector<std::vector<int>> node_shapes;
+
+  std::vector<int> dims;
+  for (unsigned i = 0, e = node_shapes_str.size(); i < e; i++) {
+    dims.clear();
+    for (const auto& dim_str : absl::StrSplit(node_shapes_str[i], ',')) {
+      // Treats empty input shape as scalar.
+      if (dim_str.empty()) continue;
+      int size;
+      absl::SimpleAtoi(dim_str, &size);
+      dims.push_back(size);
+    }
+    node_shapes.push_back(dims);
+  }
+  std::vector<string> output_nodes = absl::StrSplit(output_names, ',');
+
+  GraphDebugInfo debug_info;
+  GraphImportConfig specs;
+
+  // Set the output to the output nodes.
+  specs.outputs = output_nodes;
+
+  // Set the input values to specs.input.
+  std::vector<std::string> used_node_dtypes;
+  if (node_dtypes.empty() ||
+      (node_dtypes.size() == 1 && node_dtypes[0].empty())) {
+    // Mark all the node dtypes Invalid, so the importer can handle them by
+    // using the type from the graph.
+    used_node_dtypes.resize(node_names.size(), DataType_Name(DT_INVALID));
+  } else if (node_names.size() == node_dtypes.size()) {
+    for (const auto& dtype : node_dtypes) {
+      if (dtype.empty()) {
+        used_node_dtypes.push_back(DataType_Name(DT_INVALID));
+      } else if (dtype != DataType_Name(DT_INVALID)) {
+        used_node_dtypes.push_back(dtype);
+        // Use '' if you want to use the type from graph.
+      } else {
+        return "// error";
+      }
+    }
+    // Unmatched input node array and data type sizes.
+  } else {
+    return "// error";
```

Comment:
Can you make all these error message more specific?

Reply:
Done.